pkg/test/framework/components/istio/operator.go

```diff
@@ -569,19 +553,19 @@ func installControlPlaneCluster(i *operatorComponent, cfg Config, c cluster.Clus
 		return err
 	}
 
-	// Set the clusterName for the local cluster.
-	// This MUST match the clusterName in the remote secret for this cluster.
 	if i.environment.IsMulticluster() {
-		if i.isExternalControlPlane() || cfg.IstiodlessRemotes {
-			// enable namespace controller writing to remote clusters
-			installSettings = append(installSettings, "--set", "values.pilot.env.EXTERNAL_ISTIOD=true")
-		}
+		// Enable namespace controller writing to remote clusters
+		installSettings = append(installSettings, "--set", "values.pilot.env.EXTERNAL_ISTIOD=true")
```

Comment:
I think this will have some unintended effects. For _every_ cluster an istiod has secrets for (remote, or another primary), we'll try to do things like webhook cert patch and namespace controller:

https://github.com/istio/istio/blob/1393b4d596fd4d37abdc3d2834842c741efb11fd/pilot/pkg/serviceregistry/kube/controller/multicluster.go#L207-L208

We need a way for the control plane to differentiate.

Reply:
I see what you mean. Interestingly all three istiodless test jobs seem to work with this bug (just lucky timing I suppose?). Anyway, I added back the istiodless checks, so we still run the tests without istiodless remotes, but the postsubmit istiodless jobs seem to be passing now.
 
I suggest we merge this PR now since it includes enough progress and then we can fix the cert patch problem and figure out how to proceed with istiodless in a future PR. If you agree, please approve.

Reply:
👍  Agree. I think https://github.com/istio/istio/pull/35286 may be the solution here. Things _should_ still work, but it's not ideal.